uiowa/uiowa
# SiteNow People View Page Filtering

Resolves #3043 

Title/Position has been removed from the scope but could be included later as part of the combined filter.

This PR assumes existing sites haven't modified the view beyond the configuration allowed here: admin/config/system/sitenow-people. The update hook grabs that current config, imports the view changes and adds the site's modifications back.

Originally, this was setup so that the filter existed in the view and the configuration form only toggled the exposed status. When not exposed, this resulted in no results based on the "hardcoded" empty values for the filters. Not how I originally envisioned it, but creating/deleting the filter each time was the only way I could get it to work and keep the same order of filters.

## To Test

- `composer install && blt dsa` v2/v3 new and existing sites. default, customized, disabled SiteNow People configuration.
  - Example sites: gis.sites.uiowa.edu/people (v2, heading text, manual sort), sandbox.prod.drupal.uiowa.edu/sandbox-people (v3, custom title, path, heading, sticky/first/last)
- Import runs correctly (existing configuration is preserved, no config diff issues, and admin/structure/views/view/people should have changed from "content" render to "fields" render with rendered entity and other fields.
- As a webmaster, changing configuration and enabling/disabling filters should work as expected (admin/config/system/sitenow-people)
- As a visitor with filters enabled, you should be able to see a dropdown of person types, and filter by values contained in the view.
- Test with a pager (more than 30 rows, https://law.local.drupal.uiowa.edu/people)
- Short of going into views ui and manually changing the view, try to break the functionality.
